libraries/AC_Fence/AC_Fence.cpp

```diff
@@ -145,14 +145,20 @@ bool AC_Fence::pre_arm_check(const char* &fail_msg) const
     return true;
 }
 
-bool AC_Fence::check_fence_alt_max(const float curr_alt)
+bool AC_Fence::check_fence_alt_max()
 {
     // altitude fence check
     if (!(_enabled_fences & AC_FENCE_TYPE_ALT_MAX)) {
         // not enabled; no breach
         return false;
     }
 
+    Vector3f home;
+    if (_ahrs.get_relative_position_NED_home(home)) {
+        // we (may) remain breached if we can't update home
```

Comment:
Humm, both the variable name and the comment suggest this method returns home, but it actually returns alt above home.
Also, any reason to not simply use `get_relative_position_D_home`? That's "guaranteed" to always return a value.